### PR TITLE
Add freezed, logger, injectable and auto-router

### PR DESCRIPTION
**Pull request by pmaite88**
The merge of this PR should not be squashed

### CODE REVIEW

File: test/widget_test.dart [hunk 1]
```diff
@@ -13,7 +13,7 @@ import 'package:flutter_bootstrap/main.dart';
 void main() {
   testWidgets('Counter increments smoke test', (WidgetTester tester) async {
     // Build our app and trigger a frame.
-    await tester.pumpWidget(const MyApp());
```
Comment (pmaite88) [inline test/widget_test.dart:16]: This is what troubles me about this PR. It was a required step when I added auto-route if I remember correctly

File: lib/pages/login.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                }
+                return null;
+              },
+            ),
+            ElevatedButton(
+              onPressed: () {
+                if (_formKey.currentState!.validate()) {
+                  ScaffoldMessenger.of(context).showSnackBar(
+                    const SnackBar(
+                      content: Text('Processing Data'),
```
Comment (darrillaga) [inline lib/pages/login.dart:65]: L10n task will fix this

Comment (pmaite88) [reply lib/pages/login.dart:65]: https://crunchloop.atlassian.net/browse/FLB-7

File: lib/app_router.gr.dart [hunk 3]
```diff
@@ -0,0 +1,91 @@
+// **************************************************************************
```
Comment (darrillaga) [inline lib/app_router.gr.dart:1]: We agreed to remove generated files on a future change

**Review by darrillaga — APPROVED** [review]

**Pull request merged**